src/com/facebook/buck/apple/AppleBinaryDescription.java

```diff
@@ -198,8 +199,16 @@ private boolean shouldWrapIntoAppleDebuggableBinary(
       BuildRuleParams params,
       BuildRuleResolver resolver,
       A args) throws NoSuchBuildTargetException {
+    SourcePathResolver pathResolver = new SourcePathResolver(new SourcePathRuleFinder(resolver));
+    CxxLibraryDescription.Arg cxxDelegateArgs = CxxLibraryDescription.createEmptyConstructorArg();
+    AppleDescriptions.populateCxxLibraryDescriptionArg(
+        pathResolver,
+        cxxDelegateArgs,
+        args,
+        params.getBuildTarget());
+
     Optional<BuildRule> swiftCompanionBuildRule = swiftDelegate.createCompanionBuildRule(
-        targetGraph, params, resolver, args);
+        targetGraph, params, resolver, delegate.getDefaultCxxPlatform(), cxxDelegateArgs, args);
```

Comment:
We should not use `defaultCxxPlatform` here. I think you should use an optional of specified `cxxPlatform`, otherwise, it might work for default platform specified in the `.buckconfig` but it won't work for target with platform flavors, for example if you want to build `A#iphoneos-x86_64`, but you have `default_platform = iphonesimulator-x86_64`, you will get problem with this.

Above of this, I would argue that we don't need extra params here, `createCompanionBuildRule` simply for creating `SwiftLibrary` rule (a place holder rule without any compile steps), `SwiftLibrary` doesn't care about `cxxPlatform`. Also, `createCompanionBuildRule` forwards all inputs from `apple_binary` and `apple_library`'s `createBuildRule` method, so as long as apple description can get those cxx information, why can't Swift library description?

I haven't looked much at bridging header support, I need to look through the issue again and get back with suggestion.